File: app/src/main/java/com/studios/lucian/students/adapter/GradesListAdapter.java
package com.studios.lucian.students.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.studios.lucian.students.R;
import com.studios.lucian.students.model.Grade;

import java.util.List;

/**
 * Created with Love by Lucian and Pi on 11.04.2016.
 */
public class GradesListAdapter extends ArrayAdapter<Grade> {

    private final Context mContext;
    private final List<Grade> mStudentsList;
    private final int mResource;

    public GradesListAdapter(Context context, List<Grade> objects) {
        super(context, R.layout.item_grade, objects);
        mContext = context;
        mStudentsList = objects;
        mResource = R.layout.item_grade;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View rowView = convertView;
        if (rowView == null) {
            rowView = LayoutInflater.from(mContext).inflate(mResource, parent, false);
            ViewHolder viewHolder = new ViewHolder();

            viewHolder.textViewDate = (TextView) rowView.findViewById(R.id.textViewDate);
            viewHolder.textViewGrade = (TextView) rowView.findViewById(R.id.textViewGrade);
            viewHolder.textViewLabNr = (TextView) rowView.findViewById(R.id.textViewLabNumber);
            rowView.setTag(viewHolder);
        }

        Grade grade = getItem(position);
        if (grade != null) {
            final ViewHolder holder = (ViewHolder) rowView.getTag();
            holder.textViewGrade.setText(String.valueOf(grade.getGrade()));
            holder.textViewLabNr.setText(String.valueOf(grade.getLabNumber()));
            holder.textViewDate.setText(grade.getDate());
        }
        return rowView;
    }

    static class ViewHolder {
        public TextView textViewGrade, textViewLabNr, textViewDate;
    }
}

File: app/src/main/java/com/studios/ciprian/students/adapter/GradesListAdapter.kt
package com.studios.ciprian.students.adapter;

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.firestore.FirestoreRecyclerAdapter
import com.firebase.ui.firestore.FirestoreRecyclerOptions
import com.studios.ciprian.students.R
import com.studios.ciprian.students.model.Grade

open class GradesListAdapter(
        options: FirestoreRecyclerOptions<Grade>
) : FirestoreRecyclerAdapter<Grade, GradesListAdapter.GradesHolder>(options) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GradesHolder {
        return GradesHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_grade, parent, false))
    }

    override fun onBindViewHolder(holder: GradesHolder, position: Int, model: Grade) {
        holder.bind(model)
    }

    class GradesHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(model: Grade) {
            itemView.findViewById<TextView>(R.id.textViewGrade).text = model.grade.toString()
            itemView.findViewById<TextView>(R.id.textViewDate).text = model.date
            itemView.findViewById<TextView>(R.id.textViewLabNumber).text = model.labNumber.toString()
        }
    }
}
